### PR TITLE
Bug/sc 28924/author cannot edit review response on clark

### PR DESCRIPTION
**Pull request by mfranz1**
After several hours of banging my head on my desk, I understand the coded logic that we call ratings. Burn it with fire 🔥 and lets start anew... (the response logic was not propagating correct between child and parent components. This PR addresses those issues. A small refactor to the confirmation modal had to be done. The other modal was beautiful but I said forget it and did my own thing)

https://github.com/Cyber4All/clark-client/assets/55514493/0c0b88b9-7fa6-4488-aaf8-f67987bf6b29

### CODE REVIEW

File: src/app/cube/details/components/learning-object-ratings/learning-object-ratings.component.html [hunk 1]
```diff
@@ -1,14 +1,15 @@
 <div class="learning-object-ratings">
   <div class="overall-rating">
     <div *ngIf="ratings?.length > 0">Overall Rating</div>
-    <clark-rating-stars  *ngIf="ratings?.length > 0" [rating]="averageRating" [color]="'gold'" [size]="22" [reviewsCount]="ratings?.length"></clark-rating-stars>
+    <clark-rating-stars *ngIf="ratings?.length > 0" [rating]="averageRating" [color]="'gold'" [size]="22"
+      [reviewsCount]="ratings?.length"></clark-rating-stars>
```
Comment (mfranz1) [inline src/app/cube/details/components/learning-object-ratings/learning-object-ratings.component.html:5]: This was my formatter.

File: src/app/cube/details/components/learning-object-ratings/learning-object-ratings.component.html [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
   </div>
   <div *ngIf="ratings?.length > 0; else noRatings" class="ratings-list">
     <div *ngFor="let rating of ratings; let i = index;" class="rating-list-element">
       <a [routerLink]="['/users', rating.user.username]">
         <div class="rating-user">
-          <img [src]="userService.getGravatarImage(rating.user.email, 45)" class="user-image" alt="Picture of {{rating.user.name}}"
-          />
+          <img [src]="userService.getGravatarImage(rating.user.email, 45)" class="user-image"
+            alt="Picture of {{rating.user.name}}" />
```
Comment (mfranz1) [inline src/app/cube/details/components/learning-object-ratings/learning-object-ratings.component.html:12]: Formatter

File: src/app/cube/details/components/learning-object-ratings/learning-object-ratings.component.html [hunk 3]
```diff
@@ -23,43 +24,54 @@
       </div>
       <div *ngIf="auth.user?.emailVerified" class="options">
         <ng-container *ngIf="isRatingAuthor(i); else notOwnerOptions">
-          <span tabindex="0" aria-label="Clickable Edit Rating link" (activate)="submitEditRating(i)"><i class="far fa-pencil-alt"></i>Edit rating</span>
-          <span tabindex="0" aria-label="Clickable Delete Rating link" (activate)="submitDeleteRating(i)"><i class="far fa-trash-alt"></i>Delete rating</span>
+          <span tabindex="0" aria-label="Clickable Edit Rating link" (activate)="submitEditRating(i)"><i
+              class="far fa-pencil-alt"></i>Edit rating</span>
+          <span tabindex="0" aria-label="Clickable Delete Rating link" (activate)="submitDeleteRating(i)"><i
+              class="far fa-trash-alt"></i>Delete rating</span>
```
Comment (mfranz1) [inline src/app/cube/details/components/learning-object-ratings/learning-object-ratings.component.html:30]: Formatter

File: src/app/cube/details/details.component.ts [hunk 4]
```diff
@@ -578,9 +578,8 @@ export class DetailsComponent implements OnInit, OnDestroy {
     index: number,
   }) {
     // locate target rating and then delete the index param from the response
-    const ratingId = this.learningObject.ratings[response.index].id;
+    const ratingId = this.learningObject.ratings.ratings[response.index].id;
```
Comment (mfranz1) [inline src/app/cube/details/details.component.ts:581]: This was one of the issues

File: src/app/cube/details/details.component.ts [hunk 5]
```diff
@@ -611,43 +610,21 @@ export class DetailsComponent implements OnInit, OnDestroy {
...
-          });
-
-        if (result) {
-          this.getLearningObjectRatings();
-          this.toastService.success('Success!', 'Response deleted successfully!');
-        } else {
-          this.toastService.error('Error!', 'An error occured and your response could not be deleted');
-        }
-      } else {
+    this.ratingService
```
Comment (mfranz1) [inline src/app/cube/details/details.component.ts:613]: I removed this modal, it was not triggering as intended. Used clark-popup

File: src/app/cube/details/components/learning-object-ratings/learning-object-ratings.component.ts [hunk 6]
```diff
@@ -10,27 +10,29 @@ import { UserService } from 'app/core/user.service';
...
+  @Output() reportRating: EventEmitter<{ concern: string, index: number, comment?: string }> = new EventEmitter();
 
   showReport = false;
   reportIndex: number;
   showResponse = [];
   showEditResponse = [];
   deleteConfirmation: boolean;
+  deleteResponseConfirmation: boolean;
   deleteRatingIndex: number;
+  deleteResponseIndex: number;
```
Comment (mfranz1) [inline src/app/cube/details/components/learning-object-ratings/learning-object-ratings.component.ts:31]: Vars for clark-popup

File: src/app/cube/details/components/learning-object-ratings/learning-object-ratings.component.html [hunk 7]
```diff
@@ -23,43 +24,54 @@
...
+              <span tabindex="0">{{ rating.response[0].user.name }}</span>
             </div>
-          </a>
-          <div class="rating-date" tabindex="0">{{ rating.response[0].date | date:'shortDate' }}</div>
-          <div class="rating-text" tabindex="0">
-            {{ rating.response[0].comment }}
           </div>
-          <div class="options">
-              <ng-container *ngIf="isResponseAuthor(i); else notOwnerOptions">
-                <button tabindex="0" (activate)="openEditResponse($event, i)"><i class="far fa-pencil-alt"></i>Edit response</button>
```
Comment (mfranz1) [inline src/app/cube/details/components/learning-object-ratings/learning-object-ratings.component.html:50]: This was the actual error, we should only be passing the index, not the $event

**Review by pzalep1 — APPROVED** [review]
image

Thank you!

**Comment by stef359** [comment]
Nice work👍

**Pull request merged**